flink-table-store-core/src/main/java/org/apache/flink/table/store/table/sink/TableCommitImpl.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.flink.table.store.file.operation.FileStoreCommit;
+import org.apache.flink.table.store.file.operation.FileStoreExpire;
+import org.apache.flink.table.store.file.operation.Lock;
+import org.apache.flink.table.store.file.operation.PartitionExpire;
+
+import javax.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An abstraction layer above {@link FileStoreCommit} and {@link FileStoreExpire} to provide
+ * snapshot commit and expiration.
+ */
+public class TableCommitImpl implements TableCommit {
+
+    private final FileStoreCommit commit;
+    @Nullable private final FileStoreExpire expire;
+    @Nullable private final PartitionExpire partitionExpire;
+
+    @Nullable private List<Map<String, String>> overwritePartitions = null;
+    @Nullable private Lock lock;
+
+    public TableCommitImpl(
+            FileStoreCommit commit,
+            @Nullable FileStoreExpire expire,
+            @Nullable PartitionExpire partitionExpire) {
+        this.commit = commit;
+        this.expire = expire;
+        this.partitionExpire = partitionExpire;
+    }
+
+    @Override
+    public TableCommitImpl withOverwritten(
+            @Nullable List<Map<String, String>> overwritePartitions) {
+        this.overwritePartitions = overwritePartitions;
+        return this;
+    }
+
+    @Override
+    public TableCommitImpl withLock(Lock lock) {
+        commit.withLock(lock);
+
+        if (expire != null) {
+            expire.withLock(lock);
+        }
+
+        if (partitionExpire != null) {
+            partitionExpire.withLock(lock);
+        }
+
+        this.lock = lock;
+        return this;
+    }
+
+    @Override
+    public TableCommitImpl ignoreEmptyCommit(boolean ignoreEmptyCommit) {
+        commit.ignoreEmptyCommit(ignoreEmptyCommit);
+        return this;
+    }
+
+    @Override
+    public Set<Long> filterCommitted(Set<Long> commitIdentifiers) {
+        return commit.filterCommitted(commitIdentifiers);
+    }
+
+    @Override
+    public void commit(long identifier, List<CommitMessage> commitMessages) {
+        ManifestCommittable committable = new ManifestCommittable(identifier);
+        for (CommitMessage commitMessage : commitMessages) {
+            committable.addFileCommittable(commitMessage);
+        }
+        commit(Collections.singletonList(committable));
+    }
+
+    public void commit(List<ManifestCommittable> committables) {
+        if (overwritePartitions == null) {
+            for (ManifestCommittable committable : committables) {
+                commit.commit(committable, new HashMap<>());
+            }
+        } else {
+            ManifestCommittable committable;
+            if (committables.size() > 1) {
+                throw new RuntimeException(
+                        "Multiple committables appear in overwrite mode, this may be a bug, please report it: "
```

Comment:
How the users solve the problem that multiple committables appear in overwrite mode? This should provide users with the solution.

Reply:
This is not user face api, this is internal api.